src/js/extends.js

```diff
@@ -49,6 +51,11 @@ const extendsFn = function(superClass, subClassMethods={}) {
   let methods = {};
 
   if (typeof subClassMethods === 'object') {
+    if (typeof subClassMethods.init === 'function') {
+      log.warn('Constructor logic via init() is deprecated; please use constructor() instead.');
+      subClassMethods.constructor = subClassMethods.init;
+      delete subClassMethods.init;
```

Comment:
I wasn't 100% sure `delete`ing the `init` was the best idea here. Had a minor concern that users inspecting the prototype chain could get thrown off, but perhaps the deprecation warning is enough.